v2/robotmk/tests/test_scheduler.rs

```diff
@@ -0,0 +1,304 @@
+                        4,
+                    ),
+                    host: Host::Source,
+                },
+            ),
+            (
+                String::from("rcc_headed"),
+                SuiteConfig {
+                    robot_framework_config: RobotFrameworkConfig {
+                        robot_target: suite_dir.join("tasks.robot"),
+                        command_line_args: vec![],
+                    },
+                    execution_config: ExecutionConfig {
+                        n_attempts_max: 1,
+                        retry_strategy: RetryStrategy::Complete,
+                        execution_interval_seconds: 45,
+                        timeout: 15,
+                    },
+                    environment_config: EnvironmentConfig::Rcc(RCCEnvironmentConfig {
+                        robot_yaml_path: suite_dir.join("robot.yaml"),
+                        build_timeout: 1200,
+                        env_json_path: None,
+                    }),
+                    session_config: SessionConfig::SpecificUser(UserSessionConfig {
+                        user_name: user_name_headed.into(),
+                    }),
+                    working_directory_cleanup_config: WorkingDirectoryCleanupConfig::MaxAgeSecs(
+                        120,
+                    ),
+                    host: Host::Source,
+                },
+            ),
+            // Note: For our test, it doesn't matter if the suite can be executed on the target
+            // system. We are not checking for success. So even on systems with no Python, the test
+            // will succeed.
+            (
+                String::from("no_rcc"),
+                SuiteConfig {
+                    robot_framework_config: RobotFrameworkConfig {
+                        robot_target: suite_dir.join("tasks.robot"),
+                        command_line_args: vec![],
+                    },
+                    execution_config: ExecutionConfig {
+                        n_attempts_max: 1,
+                        retry_strategy: RetryStrategy::Complete,
+                        execution_interval_seconds: 37,
+                        timeout: 17,
+                    },
+                    environment_config: EnvironmentConfig::System,
+                    session_config: SessionConfig::Current,
+                    working_directory_cleanup_config: WorkingDirectoryCleanupConfig::MaxExecutions(
+                        4,
+                    ),
+                    host: Host::Piggyback("oink".into()),
+                },
+            ),
+        ]
+        .into(),
+    }
+}
+
+async fn run_scheduler(test_dir: &Utf8Path, config: &Config, n_seconds_run: u64) -> Result<()> {
+    let config_path = test_dir.join("config.json");
+    write(&config_path, to_string(&config)?)?;
+    let run_flag_path = test_dir.join("run_flag");
+    write(&run_flag_path, "")?;
+
+    let mut robotmk_cmd = Command::new(cargo_bin("robotmk"));
```

Comment:
I'm unsure about this. We could also expose library functions, which allow calling the scheduler directly.

Reply:
Hmm, then we should use the windows-agent to run the scheduler, no?

Reply:
Yes, we actually should, but that's more involved.,